async-nats/src/service/mod.rs

```diff
@@ -837,7 +841,11 @@ impl EndpointBuilder {
 
     /// Finalizes the builder and adds the [Endpoint].
     pub async fn add<S: ToString>(self, subject: S) -> Result<Endpoint, Error> {
-        let subject = subject.to_string();
+        let mut subject = subject.to_string();
+        if let Some(prefix) = self.prefix {
+            subject = format!("{}.{}", prefix, subject);
+        }
+        let endpoint_name = format!("{}-{}", subject, self.queue_group);
```

Comment:
I am not sure about changing the endpoint name implicitly here. First of all, that seems to be rust-specific, we don't do it in other clients. Secondly, this can be harmful when you want to e.g. get stats of a specific test, like you do in a test that's failing in CI now:
```rust
        assert!(service
            .stats()
            .await
            .get("products")
            .unwrap()
            .last_error
            .is_some());
```

This fails because endpoint name is `products-q` rather than `products`. I think this can be confusing for the user of the API. My assumption is, the endpoint name is purely specified by users and we should not interfere with it in any way.

Reply:
Yes, I needed to differentiate endpoints with the same names but different queue groups, but it should be internal name. Thx.